Modules/Loadable/Markups/MRML/vtkSlicerDijkstraGraphGeodesicPath.cxx

```diff
@@ -0,0 +1,188 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+  and was supported through CANARIE's Research Software Program, Cancer
+  Care Ontario, OpenAnatomy, and Brigham and Women�s Hospital through NIH grant R01MH112748.
+
+==============================================================================*/
+
+// Markups MRML includes
+#include "vtkSlicerDijkstraGraphGeodesicPath.h"
+
+// VTK includes
+#include <vtkFloatArray.h>
+#include <vtkPointData.h>
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+
+//------------------------------------------------------------------------------
+vtkStandardNewMacro(vtkSlicerDijkstraGraphGeodesicPath);
+
+//------------------------------------------------------------------------------
+vtkSlicerDijkstraGraphGeodesicPath::vtkSlicerDijkstraGraphGeodesicPath()
+{
+  this->UseScalarWeights = true;
+  this->RecalculateAdjacency = true;
+  this->CostFunction = COST_FUNCTION_DISTANCE;
+}
+
+//------------------------------------------------------------------------------
+vtkSlicerDijkstraGraphGeodesicPath::~vtkSlicerDijkstraGraphGeodesicPath()
+= default;
+
+//------------------------------------------------------------------------------
+void vtkSlicerDijkstraGraphGeodesicPath::PrintSelf(std::ostream &os, vtkIndent indent)
+{
+  Superclass::PrintSelf(os, indent);
+  os << indent << "CostFunction: " << this->GetCostFunctionAsString(this->CostFunction) << std::endl;
+}
+
+//------------------------------------------------------------------------------
+const char* vtkSlicerDijkstraGraphGeodesicPath::GetCostFunctionAsString(int costFunction)
+{
+  switch (costFunction)
+    {
+    case COST_FUNCTION_DISTANCE:
+      {
+      return "distance";
+      }
+    case COST_FUNCTION_ADDITIVE:
+      {
+      return "additive";
+      }
+    case COST_FUNCTION_INVERSE_SQUARED:
+      {
+      return "inverseSquared";
+      }
+    default:
+      {
+      return "";
+      }
+    }
+}
+
+//------------------------------------------------------------------------------
+const char* vtkSlicerDijkstraGraphGeodesicPath::GetCostFunctionAsHumanReadableString(int costFunction)
+{
+  switch (costFunction)
+    {
+    case COST_FUNCTION_DISTANCE:
+      {
+      return "Distance";
+      }
+    case COST_FUNCTION_ADDITIVE:
+      {
+      return "Additive";
+      }
+    case COST_FUNCTION_INVERSE_SQUARED:
+      {
+      return "Inverse squared";
+      }
+    default:
+      {
+      return "";
+      }
+    }
+}
+
+//----------------------------------------------------------------------------
+int vtkSlicerDijkstraGraphGeodesicPath::RequestData(
+  vtkInformation* vtkNotUsed(request),
+  vtkInformationVector** inputVector,
+  vtkInformationVector* outputVector)
+{
+  vtkInformation* inInfo = inputVector[0]->GetInformationObject(0);
```

Comment:
add a comment how it is related to the parent class implementation (what's the difference, why)